server/accounts.go

```diff
@@ -2846,20 +2844,26 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 	// Reset any notion of export revocations.
 	a.actsRevoked = nil
 
+	alteredScope := map[string]struct{}{}
+
 	// update account signing keys
 	a.signingKeys = nil
+	if len(ac.SigningKeys) > 0 {
+		a.signingKeys = make(map[string]jwt.Scope, len(ac.SigningKeys))
+	}
 	signersChanged := false
-	for k := range ac.SigningKeys {
-		a.signingKeys = append(a.signingKeys, k)
+	for k, scope := range ac.SigningKeys {
```

Comment:
this doesn't look right it is simply counting - it could have the same count, but a value is changed